src/go/rpk/pkg/cli/cmd/wasm/deploy_test.go

```diff
@@ -0,0 +1,214 @@
+	createTopic	func(string, *sarama.TopicDetail, bool) error
+	listTopics	func() (map[string]sarama.TopicDetail, error)
+	describeCluster	func() (brokers []*sarama.Broker, controllerID int32, err error)
+}
+
+func (a *mockAdmin) CreateTopic(
+	topic string, detail *sarama.TopicDetail, validateOnly bool,
+) error {
+	if a.createTopic != nil {
+		return a.createTopic(topic, detail, validateOnly)
+	}
+	return nil
+}
+
+func (a *mockAdmin) ListTopics() (map[string]sarama.TopicDetail, error) {
+	if a.listTopics != nil {
+		return a.listTopics()
+	}
+	return map[string]sarama.TopicDetail{}, nil
+}
+
+func (a *mockAdmin) DescribeCluster() (
+	brokers []*sarama.Broker,
+	controllerID int32,
+	err error,
+) {
+	if a.describeCluster != nil {
+		return a.describeCluster()
+	}
+	mock := make([]*sarama.Broker, 0)
+	return mock, 0, nil
+}
+
+func TestNewDeployCommand(t *testing.T) {
+	tests := []struct {
+		name		string
+		producer	func(bool, int32) (sarama.SyncProducer, error)
+		fileInfo	fileInfo
+		args		[]string
+		expectedOutput	[]string
+		expectedErr	string
+		pre		func(fs afero.Fs, fileInformation fileInfo) error
+		failSendMessage	bool
+		admin		*mockAdmin
+	}{
+		{
+			name:	"it should publish a message with correct format",
+			args:	[]string{"--description", "coprocessor description"},
+			fileInfo: fileInfo{
+				name:		"fileName.js",
+				content:	"let s = 'text'",
+			},
+			pre: func(fs afero.Fs, fileInformation fileInfo) error {
+				return createMockFile(fs, fileInformation.name, fileInformation.content)
+			},
+		}, {
+			name:	"it should fail if the file extension isn't js",
+			fileInfo: fileInfo{
+				name: "fileName.html",
+			},
+			expectedErr:	"can't deploy 'fileName.html': only .js files are supported.",
+		}, {
+			name:	"it should fail if the file doesn't exist",
+			fileInfo: fileInfo{
+				name: "fileName.js",
+			},
+			expectedErr:	"open fileName.js: file does not exist",
+		}, {
+			name:	"it should show an error if there is a error on send message",
+			fileInfo: fileInfo{
+				name: "fileName.js",
+			},
+			pre: func(fs afero.Fs, fileInformation fileInfo) error {
+				return createMockFile(fs, fileInformation.name, fileInformation.content)
+			},
+			expectedErr: "Error on deploy fileName.js, message error: kafka mock error," +
+				" please check your redpanda instance ",
+			failSendMessage:	true,
+		}, {
+			name:	"it should create a coprocessor_internal_topic if it doesn't exist",
+			fileInfo: fileInfo{
+				name: "fileName.js",
+			},
+			pre: func(fs afero.Fs, fileInformation fileInfo) error {
+				return createMockFile(fs, fileInformation.name, fileInformation.content)
+			},
+			admin: &mockAdmin{
+				listTopics: func() (map[string]sarama.TopicDetail, error) {
+					return map[string]sarama.TopicDetail{}, nil
+				},
+				createTopic: func(topic string, detail *sarama.TopicDetail, b bool) error {
+					_, exist := detail.ConfigEntries["cleanup.policy"]
+					require.True(t, exist)
```

Comment:
Don't do `require`ments inside mocked implementations. Instead, modify them to behave in different ways that allow you to verify that code works as expected.

Reply:
I tried, but if I don't make this validation into the mock function, I can't validate the correct function values